### PR TITLE
feat: Add support for the DAVerifier

### PR DESCRIPTION
**Pull request by SweeXordious**
## Overview

Closes https://github.com/celestiaorg/quantum-gravity-bridge/issues/133

~Depends on: https://github.com/celestiaorg/quantum-gravity-bridge/pull/137 to be able to create the wrappers~

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

**Comment by SweeXordious** [comment]
@adlerjohn Does it make sense to revive this PR? 
It provides a way to verify that shares were committed to in the QGB smart contract. It requires sending the data, which is not the mechanism we will use to prove inclusion for rollups, but I guess it might be good to also have the ability to verify the proofs generated inside the app in here

**Comment by SweeXordious** [comment]
@adlerjohn @evan-forbes As part of the fraud proof mechanism that will  be used, we will need to have this DAVerifier wrapper around the NMT and binary merkle tree proofs to verify the inclusion proofs correctly. 
Thus, we can revive this PR again.

File: Makefile [hunk 1]
```diff
@@ -8,6 +8,6 @@ gen: solidity-wrappers
 
 SOLIDITY_DIR = .
 SOLIDITY_SRC_DIR = $(SOLIDITY_DIR)/src
-CONTRACTS = QuantumGravityBridge.sol
+CONTRACTS = QuantumGravityBridge.sol DAVerifier.sol
```
Comment (adlerjohn) [inline Makefile:11]: The DAVerifier wrapper isn't checked in CI, only the QuantumGravityBridge wrapper. Can the former be added?

File: src/DAVerifier.sol [hunk 2]
```diff
@@ -0,0 +1,164 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.4;
```
Comment (adlerjohn) [inline src/DAVerifier.sol:2]: Bump to v0.8.19 in here and other new files.

Comment (SweeXordious) [reply src/DAVerifier.sol]: https://github.com/celestiaorg/quantum-gravity-bridge/pull/138/commits/211fc70379190996ab4001393bc73b49d784ac7b

File: Makefile (hunk 1 above)
Comment (SweeXordious) [reply Makefile:11]: [61f844b](https://github.com/celestiaorg/quantum-gravity-bridge/pull/138/commits/61f844b88328446e0f098bec12a2296bc5fcc7f4)

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -65,7 +65,7 @@ jobs:
 
       - name: Check Go wrapper match
         run: |
-          if [ "$(git diff --ignore-space-at-eol wrappers/QuantumGravityBridge.sol/wrapper.go | wc -l)" -gt "0" ]; then
+          if [ "$(git diff --ignore-space-at-eol wrappers/QuantumGravityBridge.sol/wrapper.go wrappers/DAVerifier.sol/wrapper.go | wc -l)" -gt "0" ]; then
```
Comment (adlerjohn) [inline .github/workflows/tests.yml:68]: Can this be done programmatically instead of hard-coding? E.g. by looping over `CONTRACTS`.

Comment (adlerjohn) [reply .github/workflows/tests.yml:68]: Yeah sure, follow-up PR is fine

**Review by adlerjohn — APPROVED** [review]

**Pull request merged**